pvlib/tests/iotools/test_bsrn.py

```diff
@@ -2,25 +2,123 @@
 tests for :mod:`pvlib.iotools.bsrn`
 """
 
-
 import pandas as pd
 import pytest
+import os
+from pvlib.iotools import read_bsrn, get_bsrn
+from ..conftest import (DATA_DIR, RERUNS, RERUNS_DELAY, assert_index_equal,
+                        requires_bsrn_credentials)
+
+
+@pytest.fixture(scope="module")
+def bsrn_credentials():
+    """Supplies the BSRN FTP credentials for testing purposes.
 
-from pvlib.iotools import bsrn
-from ..conftest import DATA_DIR, assert_index_equal
+    Users should obtain there own credentials as described in the `read_bsrn`
+    documentation."""
+    bsrn_username = os.environ["BSRN_FTP_USERNAME"]
+    bsrn_password = os.environ["BSRN_FTP_PASSWORD"]
+    return bsrn_username, bsrn_password
 
 
-@pytest.mark.parametrize('testfile,expected_index', [
-    ('bsrn-pay0616.dat.gz',
-     pd.date_range(start='20160601', periods=43200, freq='1min', tz='UTC')),
-    ('bsrn-lr0100-pay0616.dat',
-     pd.date_range(start='20160601', periods=43200, freq='1min', tz='UTC')),
+@pytest.fixture
+def expected_index():
+    return pd.date_range(start='20160601', periods=43200, freq='1min',
+                         tz='UTC')
+
+
+@pytest.mark.parametrize('testfile', [
+    ('bsrn-pay0616.dat.gz'),
+    ('bsrn-lr0100-pay0616.dat'),
 ])
 def test_read_bsrn(testfile, expected_index):
-    data = bsrn.read_bsrn(DATA_DIR / testfile)
+    data, metadata = read_bsrn(DATA_DIR / testfile)
+    assert_index_equal(expected_index, data.index)
+    assert 'ghi' in data.columns
+    assert 'dni_std' in data.columns
+    assert 'dhi_min' in data.columns
+    assert 'lwd_max' in data.columns
+    assert 'relative_humidity' in data.columns
+
+
+def test_read_bsrn_logical_records(expected_index):
+    # Test if logical records 0300 and 0500 are correct parsed
+    # and that 0100 is not passed when not specified
+    data, metadata = read_bsrn(DATA_DIR / 'bsrn-pay0616.dat.gz',
+                               logical_records=['0300', '0500'])
+    assert_index_equal(expected_index, data.index)
+    assert 'lwu' in data.columns
+    assert 'uva_global' in data.columns
+    assert 'uvb_reflected_std' in data.columns
+    assert 'ghi' not in data.columns
+
+
+def test_read_bsrn_bad_logical_record():
+    # Test if ValueError is raised if an unsupported logical record is passed
+    with pytest.raises(ValueError, match='not in'):
+        read_bsrn(DATA_DIR / 'bsrn-lr0100-pay0616.dat',
+                  logical_records=['dummy'])
+
+
+def test_read_bsrn_logical_records_not_found():
+    # Test if an empty dataframe is returned if specified LRs are not present
```

Comment:
seems like this comment doesn't match the actual test

Reply:
What makes you think that?

In line 67 of the same test, the index is checked if it is empty:
`    assert_index_equal(pd.DataFrame().index, data.index)`

Perhaps that I check for certain columns to be present mislead you. That is only to check that the empty dataframe has the correct columns.

Reply:
> What makes you think that?

Careless reading.  I retract my objection :)

`assert data.empty` might be a little easier to parse than `assert_index_equal(pd.DataFrame().index, data.index)`

Reply:
Hahah. I have changed it to `data.empty`, which probably would have avoided the confusion.